test/other/test_dlopen_promise.c

```diff
@@ -0,0 +1,31 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <stdio.h>
+
+#include <emscripten/emscripten.h>
+#include <emscripten/promise.h>
+
+em_promise_result_t on_fullfilled(void **result, void* data, void *handle) {
+  printf("onsuccess\n");
+  int* foo = (int*)dlsym(handle, "foo");
+  assert(foo);
+  printf("foo = %d\n", *foo);
+  assert(*foo == 42);
+  emscripten_promise_destroy((em_promise_t)data);
+  return EM_PROMISE_FULFILL;
+}
+
+em_promise_result_t on_rejected(void **result, void* data, void *value) {
+  printf("onerror %s\n", dlerror());
+  emscripten_promise_destroy((em_promise_t)data);
+  return EM_PROMISE_FULFILL;
+}
+
+int main() {
+  em_promise_t inner = emscripten_dlopen_promise("libside.so", RTLD_NOW);
+  em_promise_t outer = emscripten_promise_then(outer, on_fullfilled, on_rejected, inner);
```

Comment:
Instead of passing `inner` to the callbacks to destroy, we can just destroy it here in `main` now.

Reply:
I ran into issues where the runtime was no longer kept alive then.. but maybe your above comment will help with that.